src/utils/ui-utils.js

```diff
@@ -0,0 +1,229 @@
+
+            offset.x = rect.left + dragHandleEl.offsetLeft - eventPosition.x;
+            offset.y = rect.top + dragHandleEl.offsetTop - eventPosition.y;
+
+            $iframeDocument.on('mousemove', onMouseMove);
+            $iframeDocument.on('selectstart', cancelIFrameSelection);
+        };
+
+        var onMouseUp = function () {
+            $iframeDocument.off('mousemove', onMouseMove);
+            $iframeDocument.off('selectstart', cancelIFrameSelection);
+        };
+
+        dragHandle.on('mousedown', onMouseDown);
+        $iframeDocument.on('mouseup', onMouseUp);
+    };
+
+    var browser_prefixes = ["webkit", "moz", "ms", "o", ""];
+    var tryFullScreenPrefix = function (obj, method) {
+        var i = 0, currentMethod, type;
+        while (i < browser_prefixes.length && !obj[currentMethod]) {
+            currentMethod = method;
+            if (browser_prefixes[i] == "") {
+                currentMethod = currentMethod.substr(0, 1).toLowerCase() + currentMethod.substr(1);
+            }
+            currentMethod = browser_prefixes[i] + currentMethod;
+            type = typeof obj[currentMethod];
+            if (type != "undefined") {
+                browser_prefixes = [browser_prefixes[i]];
+                return (type == "function" ? obj[currentMethod]() : obj[currentMethod]);
+            }
+            i++;
+        }
+    };
+
+    return {
+        makeElementDraggable: makeElementDraggable,
+        makeIframeDraggable: makeIframeDraggable,
+        tryFullScreenPrefix: tryFullScreenPrefix
+    }
+};
+
+/**
+ * Utils that checks environment for compatibility with assistant
+ * @param settings
+ * @returns {{checkVisibleAreaSize: checkVisibleAreaSize, validateBrowser: validateBrowser, validatePage: validatePage}}
+ * @constructor
+ */
+var UIValidationUtils = function (settings) {
+    var document = window.document;
+    /**
+     * Check if visible area are enough to show menu.
+     * @returns boolean. True if area enough
+     */
+    var checkVisibleAreaSize = function () {
+        var viewPort = getViewPort();
+        return viewPort.height > settings.Constants.MINIMUM_VISIBLE_HEIGHT_TO_SHOW_BUTTON;
+    };
+
+    var getViewPort = function () {
+        var viewPortWidth;
+        var viewPortHeight;
+
+        viewPortWidth = window.innerWidth,
+            viewPortHeight = window.innerHeight;
+
+        return {width: viewPortWidth, height: viewPortHeight};
+    };
+
+    /**
+     * Checks if browser is valid for Adguard assistant
+     * @returns boolean. True if browser valid
+     */
+    var validateBrowser = function () {
+        return !document.documentMode || (document.documentMode > settings.Constants.MINIMUM_IE_SUPPORTED_VERSION);
+    };
+
+    /**
+     * Checks if page is valid for Adguard assistant to work here.
+     */
+    var validatePage = function () {
+        // Assistant do not work in iframes
+        if (window != top) return false;
```

Comment:
Try to use strict comparison where it's possible. Also, I see JSHint warnings here.